contracts/ccip/GovernanceCCIPReceiver.sol

```diff
@@ -3,81 +3,55 @@ pragma solidity ^0.8.20;
 
 import {Client} from "@chainlink/contracts-ccip/src/v0.8/ccip/libraries/Client.sol";
 import {CCIPReceiver} from "@chainlink/contracts-ccip/src/v0.8/ccip/applications/CCIPReceiver.sol";
-import {Test, console} from "forge-std/Test.sol";
+import {IGovernanceCCIPReceiver} from "./interfaces/IGovernanceCCIPReceiver.sol";
 
 /**
  * @title GovernanceReceiver
  * @dev A contract for receiving and executing governance proposals from Ethereum Mainnet via CCIP.
  * This contract processes cross-chain messages and executes them on the destination chain.
  */
-contract GovernanceCCIPReceiver is CCIPReceiver {
-  /// @notice The address of the sender contract on Ethereum Mainnet.
-  address public mainnetSender;
+contract GovernanceCCIPReceiver is IGovernanceCCIPReceiver, CCIPReceiver {
+  /// @inheritdoc IGovernanceCCIPReceiver
+  address public immutable mainnetSender;
 
-  /// @notice The chain selector for Ethereum Mainnet.
-  uint64 public mainnetChainSelector;
+  /// @inheritdoc IGovernanceCCIPReceiver
+  uint64 public constant mainnetChainSelector = 5009297550715157269;
 
-  /// @notice Emitted when a message is successfully executed.
-  /// @param target The target address on the destination chain.
-  /// @param payload The payload of the message.
-  event MessageExecuted(address target, bytes payload);
-  /// @notice Emitted when the mainnet sender address is updated.
-  /// @param oldSender The previous sender address.
-  /// @param newSender The new sender address.
-  event MainnetSenderUpdated(address oldSender, address newSender);
-
-  /// @dev Error thrown when an unauthorized caller tries to execute a restricted function.
-  /// @param caller The address of the unauthorized caller.
-  error Unauthorized(address caller);
-  /// @dev Error thrown when the execution of a message fails.
-  error MessageCallFailed();
-  /// @dev Error thrown when the target address is zero.
-  error TargetAddressCannotBeZero();
-  /// @dev Error thrown when the chain selector of the incoming message is invalid.
-  error InvalidChainSelector();
-
-  /**
-   * @dev Constructor to initialize the BaseGovernanceReceiver contract.
-   * @param _mainnetChainSelector The chain selector for Ethereum Mainnet.
-   */
-  constructor(
-    address _router,
-    uint64 _mainnetChainSelector,
-    address _sender
-  ) CCIPReceiver(_router) {
-    mainnetChainSelector = _mainnetChainSelector;
+  /// @dev Constructor to initialize the BaseGovernanceReceiver contract.
```

Comment:
it's still referring to the `BaseGovernanceReceiver` while it should say `GovernanceReceiver`

Reply:
done